src/openms/source/FILTERING/DATAREDUCTION/Deisotoper.cpp

```diff
@@ -38,9 +38,292 @@
 #include <OpenMS/FILTERING/DATAREDUCTION/Deisotoper.h>
 #include <OpenMS/KERNEL/MSSpectrum.h>
 #include <OpenMS/MATH/MISC/MathFunctions.h>
+#include <OpenMS/CHEMISTRY/ISOTOPEDISTRIBUTION/CoarseIsotopePatternGenerator.h>
 
 namespace OpenMS
 {
+// static
+void Deisotoper::deisotopeWithAveragineModel(MSSpectrum& spec,
+  double fragment_tolerance,
+  bool fragment_unit_ppm,
+  bool rem_low_intensity,
+  int min_charge,
+  int max_charge,
+  bool keep_only_deisotoped,
+  unsigned int min_isopeaks,
+  unsigned int max_isopeaks,
+  bool make_single_charged,
+  bool use_averagine_model,
+  bool annotate_charge,
+  bool annotate_iso_peak_count,
+  bool add_up_intensity,
+  bool used_for_open_search)
+{
+  OPENMS_PRECONDITION(spec.isSorted(), "Spectrum must be sorted.");
+
+  if (min_isopeaks < 2 || max_isopeaks < 2 || min_isopeaks > max_isopeaks)
+  {
+    throw Exception::IllegalArgument(__FILE__,
+      __LINE__,
+      OPENMS_PRETTY_FUNCTION,
+      "Minimum/maximum number of isotopic peaks must be at least 2 (and min_isopeaks <= max_isopeaks).");
+  }
+
+  if (spec.empty()) { return; }
+
+  // discard low-intensity peaks
+  if (rem_low_intensity)
```

Comment:
see https://github.com/OpenMS/OpenMS/blob/develop/src/openms/source/ANALYSIS/ID/SimpleSearchEngineAlgorithm.cpp#L227
this could shorten the code

Reply:
and maybe remove 0 intensiteis first with: https://github.com/OpenMS/OpenMS/blob/develop/src/openms/source/ANALYSIS/ID/SimpleSearchEngineAlgorithm.cpp#L192

Reply:
Yes, these classes are perfect. I think it is slightly less efficient now, but code is much shorter